### PR TITLE
Adding solo8 for micro drivers

### PR DESCRIPTION
**Pull request by jviereck**
This adds the solo8 support for dgm with the new micro drivers.

Merge only after https://github.com/open-dynamic-robot-initiative/blmc_robots/pull/55.

### CODE REVIEW

File: include/dg_blmc_robots/dgm_solo8ti.hpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ * @file dgm_solo.hpp
+ * @author Manuel Wuthrich
+ * @author Maximilien Naveau
+ * @author Julian Viereck
+ * @author Johannes Pfleging
+ * @license License BSD-3-Clause
+ * @copyright Copyright (c) 2019, New York University and Max Planck Gesellshaft.
+ */
+
+#ifndef DGM_SOLO_HH
```
Comment (MaximilienNaveau) [inline include/dg_blmc_robots/dgm_solo8ti.hpp]: `#pragma once` here

**Comment by jviereck** [comment]
@MaximilienNaveau  *ping* - can you look into the review of this one?

This one is required to land https://github.com/open-dynamic-robot-initiative/robot_properties_solo/pull/14.

File: include/dg_blmc_robots/dgm_solo8ti.hpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    /**
+     * @brief ~DemoSingleMotor is the destructor.
+     */
+    ~DGMSolo8TI();
+
+    /**
+     * @brief This function make also sure that the joint velocity do not exceed
+     * a certain value
+     */
+//    bool is_in_safety_mode();
```
Comment (MaximilienNaveau) [inline include/dg_blmc_robots/dgm_solo8ti.hpp]: to be removed if not used

File: include/dg_blmc_robots/dgm_solo8ti.hpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    void get_sensors_to_map(dynamic_graph::VectorDGMap& map);
+
+    /**
+     * @brief set_motor_controls_from_map reads the input map that contains the
+     * controls and send these controls to the hardware.
+     * @param map
+     */
+    void set_motor_controls_from_map(const dynamic_graph::VectorDGMap& map);
+
+    /**
```
Comment (MaximilienNaveau) [inline include/dg_blmc_robots/dgm_solo8ti.hpp:54]: fill in the doc? (not mandatory but would be nice)

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -4,13 +4,15 @@
...
     solo8
+    solo8ti
     solo12
     solo_simple_simu
     teststand
     stuggihop
 )
 
 set(ROBOT_CONFIG_FILES
+    "${robot_properties_solo_SOURCE_PREFIX}/config/dgm_parameters_solo8.yaml"
```
Comment (MaximilienNaveau) [inline src/CMakeLists.txt:15]: Bug?

File: src/dgm_solo8ti.cpp [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+    /** initialize the user commands */
+    ros_user_commands_.push_back(ros_node_handle.advertiseService(
+        "calibrate_joint_position",
+        &DGMSolo8TI::calibrate_joint_position_callback, this));
+
+    solo_.initialize();
+  }
+
+//  bool DGMSolo8TI::is_in_safety_mode()
```
Comment (MaximilienNaveau) [inline src/dgm_solo8ti.cpp]: remove?

**Review by MaximilienNaveau — APPROVED** [review]
Please modify the few things and you can merge.

File: src/CMakeLists.txt (hunk 4 above)
Comment (jviereck) [reply src/CMakeLists.txt:15]: Not sure if this this is a bug. In the `ROBOT_CONFIG_FILES`, you have to specify one yaml file for each robot. The robot order defined the lines above as `ROBOT_NAMES` is `solo8, solo8ti, ...`. So the `dgm_parameters_solo8.yaml` should show up twice.

Does this make sense?

**Pull request merged**